pkg/crc/machine/vsock.go

```diff
@@ -146,3 +146,21 @@ func getSSHTunnelURI() string {
 	}
 	return u.String()
 }
+
+func portFallbackWarning(ingressHTTPPort, ingressHTTPSPort uint) string {
+	var fallbackPortWarning string
+	userAssignedHTTPPort := strconv.FormatUint(uint64(ingressHTTPPort), 10)
+	userAssignedHTTPSPort := strconv.FormatUint(uint64(ingressHTTPSPort), 10)
+	fallbackPortWarningTmpl := `
+Note: Port %s is Assigned.
+You have to add this port to the application route to access
+it. like %s://myapp.apps-crc.testing:%s
+`
+	if userAssignedHTTPSPort != httpsPort {
+		fallbackPortWarning += fmt.Sprintf(fallbackPortWarningTmpl, userAssignedHTTPSPort, "https", userAssignedHTTPSPort)
+	}
+	if userAssignedHTTPPort != httpPort {
+		fallbackPortWarning += fmt.Sprintf(fallbackPortWarningTmpl, userAssignedHTTPPort, "http", userAssignedHTTPPort)
+	}
+	return fallbackPortWarning
```

Comment:
I don't really see the point of carrying this warning string all the way from `pkg/crc/machine/vsock.go` so that it can be displayed in `cmd/crc/cmd/start.go`. We could just check if ingress-http-port/ingress-https-port are set to their default values, and print the warning if not?

Reply:
> We could just check if ingress-http-port/ingress-https-port are set to their default values, and print the warning if not?

@cfergeau where you want to check it, in `cmd/start.go` ? If I do that I can't have it at the end as what we discussed previously so current implementation seemed clean to me.

Reply:
The current implementation prints the message in `cmd/start.go` and manages to print it at the end, so I don't understand your objection?

Reply:
> The current implementation prints the message in `cmd/start.go` and manages to print it at the end, so I don't understand your objection?

@cfergeau current implementation works because we carry this warning from `pkg/crc/machine/vsock.go` or you are saying we just should should check that in `pkg/crc/machine/start.go` and return from there?

Reply:
If done in `pkg/crc/machine/start.go`, I'd prefer to see expose it as `fallbackHTTPPortInUse bool; fallbackHTTPSPortInUse bool;`, and convert this to a warning in `cmd/start.go`. It's probably fine to not even do that, and check the config in `cmd/start.go`? Or does it have downsides?